### PR TITLE
Change built-in themes to use curly underlines

### LINKED ISSUE #5361: Make diagnostics messages use curly unerline in built-in themes
Currently many of the built-in themes have something like:
```toml
diagnostic = { modifiers = ["underlined"] }

error = "red"
warning = "yellow"
info = "sky"
hint = "teal"
``` 

I think it would be better to replace it with:
```toml
"diagnostic.hint" = { underline = { color = "teal", style = "curl" } }
"diagnostic.info" = { underline = { color = "sky", style = "curl" } }
"diagnostic.warning" = { underline = { color = "yellow", style = "curl" } }
"diagnostic.error" = { underline = { color = "red", style = "curl" } }

error = "red"
warning = "yellow"
info = "sky"
hint = "teal"
```

Just use curly underline for all of them and use the same color they are using for `error`, `warning`, `info`, and `hint` properties. That's what default theme is doing and it should be pretty easy to do. 

I think it should be done to all themes that  are underlining their diagnostics messages.

### PR DESCRIPTION
**Pull request by blt-r**
Fixes #5361.
I didn't touch themes that don't underline their diagnostics messages and themes that use `underline.style = "line"` instead of `modifiers = ["underlined"]`

### CODE REVIEW

**Comment by pascalkuthe** [comment]
I think this is definitely an improvement. Adding separate colored underlines for diagnostics provides additional information to the user and style wise curly also seems the most appropriate for diagnostics to me.

However I think we usually get the theme author to sign off on theme changes. It's not a hard requirement but it would be good to ping the authors of all themes changed so they have a chance to take a look at the colors you used/offer their input.

**Comment by pascalkuthe** [comment]
I think it might also be a good idea to add this to the themelint aswell (found in `xtask/src/themelint.rs`)

**Comment by krfl** [comment]
Fleet Dark is currently using `style = "line"` to match the official Fleet editor. If `curl` is the way forward for all themes, then please go ahead and change it as I do not see fleet dark included in the changes.

**Comment by blt-r** [comment]
As I said, 
> I didn't touch themes that don't underline their diagnostics messages and themes that use `underline.style = "line"` instead of `modifiers = ["underlined"]`

But now I'll add change Fleet Dark as well

**Comment by zetashift** [comment]
> Hey everyone! Do you mind if I modify your themes to make use of curly and colored underlines for diagnostic messages? (Silence will be interpreted as as "Ok")

I don't mind, nor do I think `kanagawa` prescribes underline styles. 
Thank you!

File: runtime/themes/nightfox.toml [hunk 1]
```diff
@@ -60,15 +60,15 @@
...
-"diagnostic" = { modifiers = ["underlined"] } # Diagnostics fallback style (editing area)
-"diagnostic.error" = { fg = "red" } #	Diagnostics error (editing area)
-"diagnostic.warning" = { fg = "yellow" } #	Diagnostics warning (editing area)
-"diagnostic.info" = { fg = "blue" } #	Diagnostics info (editing area)
-"diagnostic.hint" = { fg = "green" } #	Diagnostics hint (editing area)
+
+"diagnostic.warning" = { underline = { color = "yellow", style = "curl" } } #	Diagnostics warning (editing area)
+"diagnostic.error" = { underline = { color = "red", style = "curl" } } #	Diagnostics error (editing area)
+"diagnostic.info" = { underline = { color = "blue", style = "curl" } } #	Diagnostics info (editing area)
+"diagnostic.hint" = { underline = { color = "green", style = "curl" } } #	Diagnostics hint (editing area)
```
[the remark this answers is not in the document]
Comment (blt-r) [reply runtime/themes/nightfox.toml:71]: Does this happen only for this theme?

**Comment by VuiMuich** [comment]
> @enkodr @vv9k @bootradev @ChrHorn @nogden @Rinnray @Yevgnen @CptPotato @AlexanderBrevig @krfl @jbaa @IrishMaestro @PORTALSURFER @intarga @zetashift @ramojus @kvrohit @kirawi @WindSoilder @n0s4 @jhscheer @0rphee @jharrilim @raygervais @two-six @nuid32 @erasin @uncomfyhalomacro @vlmutolo @workingj @VuiMuich @p4ymak @atog @paulgraydon @kamek-pf 
> 
> 
> 
> Hey everyone! Do you mind if I modify your themes to make use of curly and colored underlines for diagnostic messages? (Silence will be interpreted as as "Ok")

Please go ahead and thanks talking this chore 🙏🏻

**Comment by paulgraydon** [comment]
Hi! Thanks a lot for doing this change, looks good to me!

**Comment by Rinnray** [comment]
> Hey everyone! Do you mind if I modify your themes to make use of curly and colored underlines for diagnostic messages? (Silence will be interpreted as as "Ok")

I don't mind at all!
There's already underline to diagnostic messages for [Doom Acario dark](https://github.com/helix-editor/helix/blob/master/runtime/themes/doom_acario_dark.toml) but maybe it was done wrong.
But thank you! 😄

**Comment by WindSoilder** [comment]
Hi, just go ahead and make the change :-D

**Comment by uncomfyhalomacro** [comment]
Sure! I won't mind ☺️

**Comment by kvrohit** [comment]
Sure! Please do. Thank you.

File: runtime/themes/monokai_pro_spectrum.toml [hunk 2]
```diff
@@ -87,7 +87,10 @@
 "diff.minus" = "red"
 
 # make diagnostic underlined, to distinguish with selection text.
-diagnostic = { modifiers = ["underlined"] }
+"diagnostic.warning" = { underline = { color = "orange", style = "curl" } }
+"diagnostic.error" = { underline = { color = "magenta", style = "curl" } } # maybe should be red?
```
Comment (LeoniePhiline) [inline runtime/themes/monokai_pro_spectrum.toml:91]: @blt-r `"magenta"` is not defined as a color in this theme, and causes problems. (See https://github.com/helix-editor/helix/issues/5249)

`"red"` must be used instead. (See `[palette]` a few lines down.)

(`"error"` using `"magenta"` is broken, too.)

Comment (blt-r) [reply runtime/themes/monokai_pro_spectrum.toml:91]: Oh, I didn't notice that magenta is not defined, just used whatever `"error"` used

Comment (LeoniePhiline) [reply runtime/themes/monokai_pro_spectrum.toml:91]: No problem. It's fixed now.

**Pull request merged**